formula/src/main/java/com/instacart/formula/plugin/Dispatcher.kt

```diff
@@ -30,10 +39,20 @@ interface Dispatcher {
             val delegate = FormulaPlugins.backgroundThreadDispatcher()
             delegate.dispatch(executable)
         }
+
+        override fun isDispatchNeeded(): Boolean {
+            val delegate = FormulaPlugins.backgroundThreadDispatcher()
+            return delegate.isDispatchNeeded()
+        }
     }
 
     /**
      * Dispatches [executable] to a thread specified by the [Dispatcher].
      */
     fun dispatch(executable: () -> Unit)
+
+    /**
+     * Returns true if dispatching event is needed
+     */
+    fun isDispatchNeeded(): Boolean
```

Comment:
Returns true if we need to dispatcher. For example, the main thread dispatcher can check if are already on main thread.